### PR TITLE
[ assets ] Fix Sass Errors

### PR DESCRIPTION
**Pull request by rogeruiz**
In the gulp asset-pipeline, any Sass related errors were being swallowed by the `gulp-size` plugin and causing it to throw an Error that wasn't being caught. Instead of working with the `gulp-size` plugin to handle errors, I have removed it as it doesn't have much value here. This PR includes line-notes for the pertinent changes to get this issue resolved.

**cc**: @maya @juliaelman

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -61,31 +59,37 @@ gulp.task('scss-lint', function (done) {
...
   if (cFlags.production) {
-    gutil.log('styles', 'Compressing styles');
+    gutil.log(gutil.colors.cyan('styles:homepage'), 'Compressing styles');
     sassStream = sass({ outputStyle: 'compressed' });
   }
 
-  stream = stream.pipe(sassStream.on('error', sass.logError))
-    .pipe(size())
+  stream = stream.pipe(sassStream)
+    .on('error', function (error) {
```
Comment (rogeruiz) [inline gulpfile.js:84]: This function now formats the error for readability and doesn't rely on the `gulp-sass` plugin.

File: gulpfile.js [hunk 2]
```diff
@@ -17,7 +17,6 @@ var filter = require('gulp-filter');
 var buffer = require('vinyl-buffer');
 var source = require('vinyl-source-stream');
 var rename = require('gulp-rename');
-var size = require('gulp-size');
```
Comment (rogeruiz) [inline gulpfile.js:20]: The root of the Error of the problems has been removed, along with all references to it below.

**Comment by rogeruiz** [comment]
The new error output is shown below

```
~/Developer/ffd-microsite fix-gulp-errors*
❯❯❯❯❯❯❯ gulp website
[11:25:57] Using gulpfile ~/Developer/ffd-microsite/gulpfile.js
[11:25:57] Starting 'clean-all'...
[11:25:57] Starting 'watch'...
[11:25:57] watch Watching assets for changes
[11:25:57] Finished 'watch' after 21 ms
[11:25:57] Finished 'clean-all' after 47 ms
[11:25:57] Starting 'build'...
[11:25:57] v1.0.0 ffd-microsite
[11:25:57]  ______ ______ _____
[11:25:57] /\  ___/\  ___/\  __-.
[11:25:57] \ \  __\ \  __\ \ \/\ \
[11:25:57]  \ \_\  \ \_\  \ \____-
[11:25:57]   \/_/   \/_/   \/____/
[11:25:57]
[11:25:57] build Building asset-pipeline
[11:25:57] Starting 'scss-lint'...
[11:25:57] Starting 'eslint'...
[11:25:58] Finished 'eslint' after 434 ms
[11:25:58] Starting 'scripts'...
[11:25:58] scripts Browserifying JavaScript assets
[11:25:58] Finished 'scripts' after 68 ms
[11:25:59] 2 issues found in /Users/rogerruiz/Developer/ffd-microsite/assets/styles/homepage.scss
[11:25:59] homepage.scss:1 [W] ColorKeyword: Color `purple` should be written in hexadecimal form as `#800080`
[11:25:59] homepage.scss:1 [W] ColorVariable: Color literals like `purple` should only be used in variable declarations; they should be referred to via variable everywhere else.
[11:25:59] Finished 'scss-lint' after 1.33 s
[11:25:59] Starting 'styles:homepage'...
[11:25:59] styles:homepage Compiling Sass assets
[11:25:59] Finished 'styles:homepage' after 9.44 ms
[11:25:59] Finished 'build' after 1.34 s
[11:25:59] Starting 'website'...
[11:25:59] website
1 of 1 draft rendered
0 future content
1 pages created
0 paginator pages created
0 tags created
0 categories created

[11:25:59] website
in 4 ms
Watching for changes in /Users/rogerruiz/Developer/ffd-microsite/{data,content,layouts,static}

[11:25:59] website
Serving pages from memory

[11:25:59] website
Web Server is available at http://localhost:1313/ (bind address 127.0.0.1)
Press Ctrl+C to stop

[11:26:04] Starting 'scss-lint'...
[11:26:05] 1 issues found in /Users/rogerruiz/Developer/ffd-microsite/assets/styles/homepage.scss
[11:26:05] homepage.scss:2 [E] Syntax Error: Invalid CSS after "... color: purple;": expected "}", was ""
[11:26:05] Finished 'scss-lint' after 898 ms
[11:26:05] Starting 'styles:homepage'...
[11:26:05] styles:homepage Compiling Sass assets
[11:26:05] styles:homepage error
 assets/styles/homepage.scss
Error: Invalid CSS after "... color: purple;": expected "}", was ""
        on line 1 of stdin
>> body { color: purple;
   ----------------------^

[11:26:05] Finished 'styles:homepage' after 5.46 ms
[11:26:09] Starting 'scss-lint'...
[11:26:10] 2 issues found in /Users/rogerruiz/Developer/ffd-microsite/assets/styles/homepage.scss
[11:26:10] homepage.scss:1 [W] ColorKeyword: Color `purple` should be written in hexadecimal form as `#800080`
[11:26:10] homepage.scss:1 [W] ColorVariable: Color literals like `purple` should only be used in variable declarations; they should be referred to via variable everywhere else.
[11:26:10] Finished 'scss-lint' after 1.15 s
[11:26:10] Starting 'styles:homepage'...
[11:26:10] styles:homepage Compiling Sass assets
[11:26:10] Finished 'styles:homepage' after 7.33 ms
[11:26:11] website
Static file changed, syncing


[11:26:15] Starting 'scss-lint'...
[11:26:16] Finished 'scss-lint' after 880 ms
[11:26:16] Starting 'styles:homepage'...
[11:26:16] styles:homepage Compiling Sass assets
[11:26:16] Finished 'styles:homepage' after 3.45 ms
[11:26:17] website
Static file changed, syncing


^C
```

**Comment by juliasolorzano** [comment]
@rogeruiz just tested and it works brilliantly - :+1: 

One thing to note, we probably want/need to add instructions for installing the 'scss-lint`gem since it is a dependency of`gulp-scss-lint`.  See note here: https://www.npmjs.com/package/gulp-scss-lint

**Comment by rogeruiz** [comment]
@juliaelman Thanks! I updated the documentation to include `scss_lint` installation instructions and added some minor semantic updates as well :100:

File: README.md [hunk 3]
```diff
@@ -27,43 +36,58 @@ You can also [download the latest release][hugo-release] and checkout the
...
-   Available tasks
-   $ gulp watch
-   Watch for changes and build the asset-pipeline
-   $ gulp [ production, no-test ] build
-   Build the asset-pipeline with optional production and no-test flags
-   $ gulp clean-all
-   Removes files and directories generated by the asset-pipeline
-   $ gulp [ production ] website
-   Runs the gulp watch and hugo serve
+❯❯❯❯❯❯❯ gulp
```
Comment (juliasolorzano) [inline README.md]: Small change. Either change to `$ gulp` for consistency or remove the `>>>>>` entirely.

**Comment by rogeruiz** [comment]
Fixed! @juliaelman

**Comment by juliasolorzano** [comment]
+1 thank you for your service!

**Pull request merged**